enhancements/sdk-fbc-bundle-upgrade.md

```diff
@@ -0,0 +1,277 @@
+
+    - It can also be seen in the example `Subscription` that we are able to get the channel that is used in a traditional OLM installation by retrieving the value in the `spec.channel` field.
+
+    - Using these values we can render the index image used in the `CatalogSource` and search for channel entries that match the channel that is specified in the `Subscription`. To infer the latest CSV that was installed we can check the channel entries for an entry that does not appear in a replaces field. This latest CSV obtained will be the edge to upgrade from.
+
+2. For example, given the following channel entries in an index image:
+
+    Example channel entries:
+
+    ```json
+    {
+      "schema": "olm.channel",
+      "name": "alpha",
+      "package": "hello-kubernetes",
+      "entries": [
+          {
+              "name": "hello-kubernetes.v0.0.1"
+          },
+          {
+              "name": "hello-kubernetes.v0.0.2",
+              "replaces": "hello-kubernetes.v0.0.1"
+          }
+        ]
+      }
+    ```
+
+    - It can be inferred that the most recently installed CSV would be `hello-kubernetes.v0.0.2` as it does not appear in a channel entry replaces field.
+
+3. After obtaining the CSV name from the above step, Operator SDK will insert an entry into the Entries field of the Channel object which looks like the following:
+
+    ```go
+        declarativeconfig.ChannelEntry{
+        Name:     “hello-kubernetes.v0.0.3”,
+        Replaces:  “hello-kubernetes.v0.0.2”,
+    }
+    ```
+
+    - From this example, `hello-kubernetes.v0.0.3` will be the CSV that we will be upgrading to from `hello-kubernetes.v0.0.2`.
+
+4. This entry will be added to the rendered index image declarative config and the upgraded bundle will also be appended to the index image declarative config.
+
+5. Furthermore, Operator SDK will convert the index image declarative config to a string format, ensure that the rendered declarative config is a valid FBC and serve the validated FBC during the ephemeral registry pod creation.
+
+- Note: Additionally, during `run bundle-upgrade` in this use-case, Operator SDK adds annotations such as bundle image, index image and injected bundles to the `CatalogSource` annotations which will be used for the subsequent upgrades without having to infer from the channel entries of the index image declarative config.
+
+- Now let's consider upgrading from `hello-kubernetes.v0.0.3` to the next version `hello-kubernetes.v0.0.4` from the above example:
+  
+  - For this case, the annotations would have been written out to `CatalogSource` during the previous upgrade to `v0.0.3`.
+
+  - We obtain the index image from CatalogSource's `spec.image` field and render it.
+
+  - We infer that the upgrade edge is `hello-kubernetes.v0.0.3` from the bundle image annotation on the `CatalogSource` and also obtain the injected bundles from the annotation which currently only has `hello-kubernetes.v0.0.3`.
```

Comment:
We would infer that the latest upgrade edge from the index image is `hello-kubernetes.v0.0.2`, right? This will then get replaced by the first element of the injected bundles annotation? It's also worth mentioning that every element in the injected bundles will get upgraded to its subsequent element.

Reply:
```suggestion
  - We infer that the upgrade edge is `hello-kubernetes.v0.0.2` from the bundle image annotation on the `CatalogSource` and also obtain the injected bundles from the annotation which currently only has `hello-kubernetes.v0.0.3`. Every CSV in the injected bundles array will get replaced by it's next element, creating a linear  upgrade graph. 
```